sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -176,11 +176,6 @@ int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev, bool full_reset)
 	hda_dsp_ctrl_misc_clock_gating(sdev, false);
 
 	if (full_reset) {
-		/* clear WAKESTS */
-		snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, SOF_HDA_WAKESTS,
-					SOF_HDA_WAKESTS_INT_MASK,
-					SOF_HDA_WAKESTS_INT_MASK);
-
```

Comment:
should remove the later one, and keep this and move it out of if (full_reset), above.

Reply:
@keyonjie You are partly right, keep this one and remove the later one does work. 

but move it out will leads to problems:
if full_reset is false, that means the wake status cleared but no actual reset.
because after reset, the wake status register will be set by the hardware and the codec_mask is from there.

So I will change the PR to: 
clear the WAKESTS and reset , also set codec_mask to 0; then after reset, the codec_mask will read  again from the register. this will handle the case when suspend, and then the codec has been changed .

Reply:
@keyonjie A update: the later one can't be removed, or the device can't suspend after the first suspend.